src/libs/actions/Report.js

```diff
@@ -727,15 +727,20 @@ function fetchAllReports(
                 return;
             }
 
-            // The string cast here is necessary as Get rvl='chatList' may return an int
-            reportIDs = String(response.chatList).split(',');
+            // The cast here is necessary as Get rvl='chatList' may return an int or Array
+            reportIDs = String(response.chatList)
+                .split(',')
+                .filter(_.identity);
```

Comment:
I'm not too sure what this `filter(_.identity)` does here? Do we need it?

Reply:
`.filter(_.identity)` removes empty/null values from the resulting array. 
I thought it's clearer than `.filter(report => report)`

This is related to the bug I've commented about in the other thread: https://github.com/Expensify/Expensify.cash/pull/2575/files#r621584917

When `response.chatList` is an empty array `[]` (After creating an account for example) `String(response.chatList)` would return `['']` - an Array containing one empty string. Then the check below `reportIDs.length > 0` would redirect flow to `fetchChatReportsByIDs` and would not actually fetch anything. The LHN stays empty and Concierge chat is never created (Unless something else creates such a chat at a different time?)

Reply:
Ok, I think it's more common for us to use `filter(report => report)` or use `_.filter()` on it's own to remove bad values, but this isn't necessarily incorrect.

<img width="283" alt="Screen Shot 2021-04-29 at 7 13 48 AM" src="https://user-images.githubusercontent.com/32969087/116591425-82656a80-a8ba-11eb-8ac9-dc2aa13f3be8.png">

Reply:
I came across this too, and I think it should be changed as it's not intuitive code (it's overly clever). Why would there be empty/null values? In what case would that ever happen?

Reply:
> This produced a bug when `response.chatList` is empty `[]`. It would assign `['']` to report IDs, and the check ` if (reportIDs.length)` would be positive even though there are no reports
> 
> You can try this to see the outcome is `['']`
> 
> ```js
> console.log(String([]).split(',')) // prints ['']
> ```

`response.chatList` is `[]` when you create a new account 
when you convert that to a string `String(response.chatList)` and then split it `.split(',')`
you get `['']` and then the Concierge chat is not created

Reply:
OK, thanks for explaining that! I think I have some ideas to clean this up.